secure_message/validation/domain.py

```diff
@@ -169,10 +169,10 @@ def check_variables_set_and_not_set(self, data):   # NOQA pylint:disable=no-self
         """Check sent and read date not set and that from and survey are set"""
         if 'msg_from' not in data or not data['msg_from']:
```

Comment:
This should be:

```Python
try:
    [data[k] for k in ['msg_from', 'survey']]
except KeyError as e:
	logger.exception(f"{e.args[0]} field missing')
else:
    return data
```

Reply:
Not quite, this this does not detect if a msg_from or survey field is present but with no value

Reply:
When would a msg_from ever be empty? A message is always sent by a user, isn't it?

Reply:
msg_from and survey are supplied by the client code and may not be present